devito/finite_difference.py

```diff
@@ -163,14 +163,13 @@ def first_derivative(*args, **kwargs):
     # Stencil positions for non-symmetric cross-derivatives with symmetric averaging
     if side == right:
         ind = [(dim + i * diff) for i in range(-int(order / 2) + 1 - (order % 2),
-                                               int((order + 1) / 2) + 2 - (order % 2))]
+                                               int(order / 2) + 2)]
```

Comment:
O, it looks like this isn't quite as harmless as it looks, since it breaks the first derivative at first order, ie.
```
In [1]: from devito import *
In [2]: u = DenseData(name='u', shape=(4, 5))
In [3]: u.dx
Out[3]: 0
```
where `master` outputs `-u(x, y)/h + u(x + h, y)/h`. This is unfortunately also missed by our unit test that do not test for `order=1`, which is bad!